internal/backends/winit/lib.rs

```diff
@@ -31,25 +33,26 @@ mod renderer {
         type Canvas: WinitCompatibleCanvas;
         const NAME: &'static str;
 
-        fn new(
-            window_adapter_weak: &Weak<dyn WindowAdapter>,
-            #[cfg(target_arch = "wasm32")] canvas_id: String,
-        ) -> Self;
+        fn new(window_adapter_weak: &Weak<dyn WindowAdapter>) -> Self;
 
-        fn create_canvas(&self, window_builder: winit::window::WindowBuilder) -> Self::Canvas;
+        fn create_canvas(
+            &self,
+            window: &dyn raw_window_handle::HasRawWindowHandle,
+            display: &dyn raw_window_handle::HasRawDisplayHandle,
```

Comment:
These could be combined into one `impl` as well, but I think for the public API of the renderer later we'll end up with `dyn` anyway - so I went for two dyn.